content/en/continuous_integration/troubleshooting.md

```diff
@@ -3,25 +3,37 @@ title: Troubleshooting CI Visibility
+3. If you still don't see any results, [contact Support][3] and we'll help you troubleshoot.
 
 ### I instrumented my tests but I don't see any data in Datadog
 
-1. Check if you see any results in the [Test Runs][4] section. In the [Tests][5] section, only tests that contain git information will be shown.
-2. If you are executing your tests in a container, make sure you are forwarding the required CI provider environment variables to them. Check out the [Running tests inside a container][6] section for more information.
-3. For all languages except Swift, make sure the Datadog Agent is running in the host where tests are run (accessible at `localhost:8126`), or if accessible on another hostname and/or port, make sure you run your tests with the appropriate hostname set in the `DD_AGENT_HOST` and the appropriate port in `DD_AGENT_PORT` environment variables.
+1. Check the _Compatibility_ section of the page of the language you are instrumenting, and make sure the testing framework you are using is supported.
+2. Check if you see any test results in the [Test Runs][4] section. If you do see results there, but not in the [Tests][5] section, Git information  might be missing. If this is this case, read the next section to troubleshoot it.
+3. For all languages (except [Swift][7]), make sure the Datadog Agent is running in the host where tests are run (accessible at `localhost:8126`), or if accessible on another hostname and/or port, make sure you run your tests with the appropriate agent hostname set in the `DD_AGENT_HOST` and the appropriate port in `DD_TRACE_AGENT_PORT` environment variables. You can activate [debug mode][8] in the tracer to check if it's being able to connect to the agent.
 4. If you still don't see any results, [contact Support][3] and we'll help you troubleshoot.
 
+### I see my tests in the "Test Runs" tab, but not in the "Tests" tab
+
+This is most probably due to missing Git metadata (repository, commit and/or branch). To confirm this is the case, open a test execution on the [Test Runs][4] section, and check that there is no `git.repository_url`, `git.commit.sha` or `git.branch`. If these tags are not populated, nothing will show on the [Tests][5] section.
+
+1. Tracers will first try to fetch Git metadata using the local `.git` folder by executing `git` commands. This is the preferred approach, as it will populate all Git metadata fields, including commit message, author and committer information.
+2. If the `.git` folder is not present, or the `git` binary is not installed, tracers will fallback to use the environment variables set by the CI provider (as listed in the [Running tests inside a container][6] section) to collect Git information. This will populate, at least, repository, commit hash and branch information.
+3. If no CI provider environment variables are found, tests results will be sent with no Git metadata.
```

Comment:
This describes what the software does, but not what the reader should do to troubleshoot. What should they check or fix? If they find themselves in step 3, does that mean they just have to live with no data on the Tests tab?

Reply:
We are currently working on a way for the user to be able to manually define Git data if (1) and (2) are impossible to do on their setup, which would replace this last step.

The reason to explain how the tracer works is for them to be able to make changes on their setup to allow the tracer to automatically detect Git data. We could rephrase this in a more "troubleshooting" way, so instead of saying what the tracer will do, we can say what they need to do (for example, "make sure the `.git` folder is present and the `git` binary is installed and in `$PATH` or something like that)

Reply:
I agree that telling them how the tracer works is valuable here. How about these rewordings to make it clearer what they can do? Let me know what you think.

Reply:
Just left a small comment on the second suggestion, otherwise LGTM

Reply:
Great! I committed my first suggestion and your second suggestion, and will 🚢